lib/dialects/postgres/data-types.js

```diff
@@ -478,7 +478,7 @@ module.exports = BaseTypes => {
 
       if (this.type instanceof BaseTypes.ENUM) {
         castKey = `${Utils.addTicks(
-          Utils.generateEnumName(options.field.Model.getTableName(), options.field.fieldName),
+          Utils.generateEnumName(options.field.tableName || options.field.Model.getTableName(), options.field.fieldName),
```

Comment:
Isn't `options.table.tableName` available here? It seems odd for attribute to keep `tableName`

Reply:
I don't think i get the `table` object in options ... i will check once again